srecord/output/file/vhdl_textio.cc

```diff
@@ -0,0 +1,154 @@
+#include <srecord/record.h>
+#include <srecord/progname.h>
+
+
+srecord::output_file_vhdl_textio::output_file_vhdl_textio(
+    const std::string &a_file_name)
+:
+    srecord::output_file(a_file_name)
+{
+}
+
+
+srecord::output::pointer
+srecord::output_file_vhdl_textio::create(const std::string &a_file_name)
+{
+    return pointer(new srecord::output_file_vhdl_textio(a_file_name));
+}
+
+
+void
+srecord::output_file_vhdl_textio::command_line(srecord::arglex_tool *cmdln)
+{
+    if (cmdln->can_get_number())
+    {
+        int in_width = cmdln->get_number("input_width");
+        if (in_width % 8)
+        {
+            warning("input width (%d) is not a multiple of 8, rounding up!\n",
+                    in_width);
+            in_width += in_width % 8;
+        }
+
+        int out_width = in_width;
+        if (cmdln->can_get_number())
+        {
+            out_width = cmdln->get_number("output_width");
+            if (out_width > in_width)
+            {
+                warning("output width (%d) is greater than input width (%d)!\n",
+                        out_width, in_width);
+                out_width = in_width;
+            }
+        }
+        consume_bytes_per_word = (in_width + 7) / 8;
+        gen_bits_per_word = out_width;
+    }
+}
+
+void
+srecord::output_file_vhdl_textio::append_bits(srecord::record::data_t data,
+                                              size_t nbits, std::string &line)
+{
+    for (srecord::record::data_t mask = 1 << (nbits - 1);
+        mask;
+        mask = mask >> 1)
+    {
+        line += (data & mask) ? "1" : "0";
+    }
+}
+
+void
+srecord::output_file_vhdl_textio::write(const srecord::record &record)
+{
+
+    switch (record.get_type())
+    {
+    case srecord::record::type_unknown:
+    case srecord::record::type_header:
+    case srecord::record::type_data_count:
+    case srecord::record::type_execution_start_address:
+        // Ignore.
+        break;
+    case srecord::record::type_data:
+        //
+        // Make sure the data is aligned.
+        //
+        if ((record.get_address() % consume_bytes_per_word) ||
+            (record.get_length() % consume_bytes_per_word))
+            fatal_alignment_error(consume_bytes_per_word);
+
+        for (size_t j = 0; j < record.get_length(); j += consume_bytes_per_word)
+        {
+            std::string line;
+            unsigned num_bits = gen_bits_per_word;
+            size_t consume_bits = num_bits % 8U;
+            if (!consume_bits) consume_bits = 8U;
+            for (unsigned k = consume_bytes_per_word; k; --k)
+            {
+                append_bits(record.get_data(j + k - 1), consume_bits, line);
+                consume_bits = 8U;
+            }
+            put_string(line.append("\n"));
+        }
+        break;
+    }
+}
+
+
+bool
+srecord::output_file_vhdl_textio::preferred_block_size_set(int nbytes)
+{
+    // this looks really wrong, from vhdl.cc ???
+    if (nbytes > 1 || nbytes > record::max_data_length)
```

Comment:
Suggest  as per above

`if (nbytes > 1 || nbytes > record::max_data_length)`

Reply:
Do you fix it int vhdl.cc?

Reply:
Yes, I have a bugfix branch. Just checking the others before I push.